blockio-sim/src/System/FS/BlockIO/Sim.hs

```diff
@@ -110,6 +124,21 @@ simTryLockFile hfs path lockmode =
         fsLimitation  = False
       }
 
+-- | @'simCreateHardLink' hfs source target@ creates a simulated hard link for the
+-- @source@ path at the @target@ path. The hard link is simulated by simply
+-- copying the source file to the target path.
```

Comment:
Worth adding a comment here that this is not a fully correct implementation of hard links. It works fine for files that are not modified, but it is not a sufficient simulation if one modifies the files: the changes would need to be visible via both/all file names.

Of course it's fine for our purposes, since we only hard link immutable files.

Reply:
Good point, if we wanted to have proper hard links we would have to bake that into `fs-sim`... I'll put a note on the function for now that it is only safe to use for immutable files

Reply:
I'm also adding a TODO about properly supporting hard links in `fs-sim`

Reply:
Done in 21696ad67cdf006e8b5558a427af909a070cfee4